### PR TITLE
Avoid adding credit when gift card redemption fails

### PR DESCRIPTION
**Pull request by LeonardoGodoy**
## [FM-2055](https://firstleaf.atlassian.net/browse/FM-2055)

This prevents invalid gift card in the database from being partially redeemed.
We were adding the credit to the users account but not updating the record to redeemed when it was invalid in the database.
Probably caused by validations added after the creation of these records.

### Changelog:
- Added transaction to make sure credit wont be added unless we mark the gift card as redeemed

### CODE REVIEW

File: app/models/spree/virtual_gift_card.rb [hunk 1]
```diff
@@ -45,7 +47,8 @@ def redeem(redeemer)
       action_originator: self,
       category: store_credit_category,
     )
-    update( redeemed_at: Time.zone.now, redeemer: redeemer )
+
+    update_status
```
Comment (jvalentino90) [inline app/models/spree/virtual_gift_card.rb:51]: update_status at this time would be only set as true right? because otherwise an exception was raised

**Review by jvalentino90 — APPROVED** [review]
LGTM

File: app/models/spree/virtual_gift_card.rb (hunk 1 above)
Comment (LeonardoGodoy) [reply app/models/spree/virtual_gift_card.rb:51]: Yes, you're right.
I put it cause one of the existing tests was expecting it to return true

Comment (LeonardoGodoy) [reply app/models/spree/virtual_gift_card.rb:51]: what do you think would be a better approach?

**Pull request merged**